src/libmongoc/src/mongoc/mongoc-openssl.c

```diff
@@ -613,32 +784,33 @@ _mongoc_ocsp_tlsext_status_cb (SSL *ssl, void *arg)
    }
 
    /* Get the stapled OCSP response returned by the server */
-   len = SSL_get_tlsext_status_ocsp_resp (ssl, &r);
-   stapled_response = !!r;
+   len = SSL_get_tlsext_status_ocsp_resp (ssl, &resp_data);
+   stapled_response = !!resp_data;
    if (stapled_response) {
       /* obtain an OCSP_RESPONSE object from the OCSP response */
-      if (!d2i_OCSP_RESPONSE (&resp, &r, len)) {
+      if (!d2i_OCSP_RESPONSE (&resp, &resp_data, len)) {
          MONGOC_ERROR ("Failed to parse OCSP response");
          ret = OCSP_CB_ERROR;
          GOTO (done);
       }
    } else {
-      MONGOC_DEBUG ("Server does not contain a stapled response");
-      must_staple = X509_get_ext_d2i (peer, NID_tlsfeature, 0, 0) != NULL;
+      TRACE ("%s", "Server does not contain a stapled response");
+      must_staple = _get_must_staple (peer);
       if (must_staple) {
          MONGOC_ERROR ("Server must contain a stapled response");
          ret = OCSP_CB_REVOKED;
          GOTO (done);
       }
 
-      if (!(resp = _contact_ocsp_responder (id, peer))) {
+      if (opts->disable_endpoint_check ||
```

Comment:
Reach out to a responder unless the user set tlsDisableOCSPEndpointCheck in the URI.